### PR TITLE
Split into binary and library

### PR DESCRIPTION
**Pull request by Cxarli**
Fixes https://github.com/SUPERAndroidAnalyzer/super/issues/124

Happy Hacktoberfest!

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -1,79 +1,36 @@
 //! SUPER Android Analyzer
-
-#![forbid(deprecated, overflowing_literals, stable_features, trivial_casts, unconditional_recursion,
```
Comment (Razican) [inline src/main.rs]: Why have all these lints been removed?

File: src/main.rs [hunk 2]
```diff
@@ -1,79 +1,36 @@
...
-#[macro_use]
-extern crate lazy_static;
-extern crate rustc_serialize;
-extern crate open;
-extern crate bytecount;
-extern crate handlebars;
+extern crate super_analyzer;
+
+
+pub extern crate clap;
```
Comment (Razican) [inline src/main.rs]: Why are we now using `pub` modifiers for extern crates?

File: src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,457 @@
...
+    single_match_else, string_add, string_add_assign, unicode_not_nfc, unseparated_literal_suffix,
+    use_debug, wrong_pub_self_convention, doc_markdown)]
+// Allowing these at least for now.
+#![allow(missing_docs_in_private_items, unknown_lints, print_stdout, stutter, option_unwrap_used,
+    result_unwrap_used, integer_arithmetic, cast_possible_truncation, cast_possible_wrap,
+    indexing_slicing, cast_precision_loss, cast_sign_loss)]
+
+
+#[macro_use]
+pub extern crate clap;
```
Comment (Razican) [inline src/lib.rs]: Why are we using `pub` modifier in extern crates?

File: src/criticality.rs [hunk 4]
```diff
@@ -0,0 +1,55 @@
+use std;
```
Comment (Razican) [inline src/criticality.rs:4]: Is there a particular reason to move this to its own module? I think it's not a bad change, but would be great if you could ellaborate. Also, some module-level documentation would be appreciated.

**Review by Razican — CHANGES_REQUESTED** [review]
Hi! Thanks for the contribution!

It looks great overall, but please, check my comments, as there are a couple of things that need clarifying.

**Comment by Razican** [comment]
I updated the base branch (pull request should be done to `develop`), and seems you should rebase, since some changes have created conflicts.

File: src/criticality.rs (hunk 4 above)
Comment (Cxarli) [reply src/criticality.rs:4]: I did not change the actual code, and it seemed that `std` was needed as an include, so I did that.

File: src/main.rs (hunk 1 above)
Comment (Cxarli) [reply src/main.rs]: They are moved to `lib.rs` if I recall correctly. Thinking of it, they should actually stay in `main.rs` too. I'll put them back in a bit

**Comment by Cxarli** [comment]
Should be fixed now :smile: The useless `pub`'s have been removed, the lints are back in `main.rs`. But, for some reason, the `use std;` in `criticality.rs` is still necessary.

**Review by Razican — APPROVED** [review]

**Comment by Razican** [comment]
Looks good to me, I will check that `std` use with more time. If it passes all tests I will merge it. Thank you for your help!

**Comment by Razican** [comment]
Seems that tests are failing with some features, could you re-check?

```
error[E0432]: unresolved import `Criticality`
   --> src/static_analysis/code.rs:662:9
    |
662 |     use Criticality;
    |         ^^^^^^^^^^^ no `Criticality` in the root. Did you mean to use `criticality`?

error[E0432]: unresolved import `Criticality`
   --> src/config.rs:622:9
    |
622 |     use Criticality;
    |         ^^^^^^^^^^^ no `Criticality` in the root. Did you mean to use `criticality`?

error[E0432]: unresolved import `Criticality`
   --> src/lib.rs:340:9
    |
340 |     use Criticality;
    |         ^^^^^^^^^^^ no `Criticality` in the root. Did you mean to use `criticality`?

error: unused import: `std::str::FromStr`
   --> src/lib.rs:341:9
    |
341 |     use std::str::FromStr;
    |         ^^^^^^^^^^^^^^^^^
    |
note: lint level defined here
   --> src/lib.rs:13:32
    |
13  | #![deny(unused_qualifications, unused, unused_attributes)]
    |                                ^^^^^^
    = note: #[deny(unused_imports)] implied by #[deny(unused)]
```

**Comment by Cxarli** [comment]
No idea what went wrong there, let me investigate.
**Update:** Should be fixed

**Comment by Razican** [comment]
Hey, I merged your other pull request and seems it created conflicts with this. Could you solve them, please?

**Comment by Cxarli** [comment]
It seems they are solved already, so I'll close my branch.

**Pull request merged**